src/doc/src/reference/unstable.md

```diff
@@ -260,6 +260,14 @@ build-dir = "out"
 
 The path to where internal files used as part of the build are placed.
 
+This option supports path templating.
+
+Avaiable template variables:
+* `{workspace-root}` resolves to root of the current workspace.
+* `{cargo-cache-home}` resolves to `CARGO_HOME`
+* `{workspace-manifest-path-hash}` resolves to a hash of the manifest path
+   * This is split into 2 nested directories where the first dir is the first 2 hex chars and the inner dir is the remaining chars. (ie. `34/f9d02eb8411c05`)
```

Comment:
FYI these are implementation details that we do not guarantee.

The second I wouldn't even specify.